tests/Doctrine/Tests/ORM/Functional/Ticket/GH6823Test.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH6823Test extends OrmFunctionalTestCase
+{
+    public function testCharsetCollationWhenCreatingForeignRelations(): void
+    {
+        if (! $this->_em->getConnection()->getDatabasePlatform() instanceof MySQLPlatform) {
+            self::markTestSkipped('This test is useful for all databases, but designed only for mysql.');
+        }
+
+        $sql = $this->_schemaTool->getCreateSchemaSql(
+            [
+                $this->_em->getClassMetadata(GH6823User::class),
+                $this->_em->getClassMetadata(GH6823Group::class),
+                $this->_em->getClassMetadata(GH6823Status::class),
+            ]
+        );
+
+        self::assertCount(5, $sql);
+
+        foreach ($sql as $query) {
+            $this->_em->getConnection()->executeQuery($query);
+        }
+
+        self::assertEquals('CREATE TABLE user (id VARCHAR(255) NOT NULL, group_id VARCHAR(255) CHARACTER SET ascii DEFAULT NULL COLLATE `ascii_general_ci`, status_id VARCHAR(255) CHARACTER SET latin1 DEFAULT NULL COLLATE `latin1_bin`, INDEX IDX_8D93D649FE54D947 (group_id), INDEX IDX_8D93D6496BF700BD (status_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_bin` ENGINE = InnoDB', $sql[0]);
+        self::assertEquals('CREATE TABLE `group` (id VARCHAR(255) NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET ascii COLLATE `ascii_general_ci` ENGINE = InnoDB', $sql[1]);
+        self::assertEquals('CREATE TABLE status (id VARCHAR(255) CHARACTER SET latin1 NOT NULL COLLATE `latin1_bin`, PRIMARY KEY(id)) DEFAULT CHARACTER SET koi8r COLLATE `koi8r_bin` ENGINE = InnoDB', $sql[2]);
+        self::assertEquals('ALTER TABLE user ADD CONSTRAINT FK_8D93D649FE54D947 FOREIGN KEY (group_id) REFERENCES `group` (id)', $sql[3]);
+        self::assertEquals('ALTER TABLE user ADD CONSTRAINT FK_8D93D6496BF700BD FOREIGN KEY (status_id) REFERENCES status (id)', $sql[4]);
+    }
+}
+
+/**
+ * @Entity
+ * @Table(name="user", options={"charset"="utf8mb4", "collate"="utf8mb4_bin"})
```

Comment:
Looks like this `options` attribute is not documented :grimacing: 
Shouldn't it be `collation` here? Where can I find a list of valid values?

Reply:
Changed it to `collation`.

The "docs" are here: https://github.com/doctrine/dbal/blob/719663b15983278227669c8595151586a2ff3327/src/Platforms/AbstractMySQLPlatform.php#L455-L464